src/main/kotlin/com/sipios/springsearch/CriteriaParser.kt

```diff
@@ -9,9 +9,9 @@ import org.springframework.data.jpa.domain.Specification
 /**
  * Class used to parse a search query string and create a specification
  */
-class CriteriaParser<T> {
+class CriteriaParser<T>(caseSensitiveFlag: Boolean) {
```

Comment:
I would pass around a reference to the initial annotation (or a context derived from it at least)

Reply:
I passed around the reference to the initial annotation